unit_test/test_spdm_requester/finish.c

```diff
@@ -3682,6 +3812,209 @@ void libspdm_test_requester_finish_case23(void **state)
+        m_libspdm_use_dhe_algo;
+    spdm_context->connection_info.algorithm.aead_cipher_suite =
+        m_libspdm_use_aead_algo;
+
+#if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
+    spdm_context->connection_info.peer_used_cert_chain[0].buffer_size =
+        data_size;
+    libspdm_copy_mem(spdm_context->connection_info.peer_used_cert_chain[0].buffer,
+                     sizeof(spdm_context->connection_info.peer_used_cert_chain[0].buffer),
+                     data, data_size);
+#else
+    libspdm_hash_all(
+        m_libspdm_use_hash_algo,
+        data, data_size,
+        spdm_context->connection_info.peer_used_cert_chain[0].buffer_hash);
+    spdm_context->connection_info.peer_used_cert_chain[0].buffer_hash_size =
+        libspdm_get_hash_size(m_libspdm_use_hash_algo);
+    libspdm_get_leaf_cert_public_key_from_cert_chain(
+        m_libspdm_use_hash_algo,
+        spdm_context->connection_info.algorithm.base_asym_algo,
+        data, data_size,
+        &spdm_context->connection_info.peer_used_cert_chain[0].leaf_cert_public_key);
+#endif
+    spdm_context->connection_info.peer_used_cert_chain_slot_id = 0;
+
+    /* Set HANDSHAKE_IN_THE_CLEAR_CAP to 0*/
+    spdm_context->connection_info.capability.flags &=
+        ~SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_HANDSHAKE_IN_THE_CLEAR_CAP;
+    spdm_context->local_context.capability.flags &=
+        ~SPDM_GET_CAPABILITIES_REQUEST_FLAGS_HANDSHAKE_IN_THE_CLEAR_CAP;
+
+    session_id = 0xFFFFFFFF;
+    session_info = &spdm_context->session_info[0];
+    spdm_context->last_spdm_request_session_id_valid = true;
+    spdm_context->last_spdm_request_session_id = session_id;
+    libspdm_session_info_init(spdm_context, session_info, session_id, false);
+    hash_size = libspdm_get_hash_size(m_libspdm_use_hash_algo);
+    libspdm_set_mem(m_libspdm_dummy_buffer, hash_size, (uint8_t)(0xFF));
+    libspdm_secured_message_set_response_finished_key(
+        session_info->secured_message_context, m_libspdm_dummy_buffer,
+        hash_size);
+    libspdm_secured_message_set_session_state(
+        session_info->secured_message_context,
+        LIBSPDM_SESSION_STATE_HANDSHAKING);
+
+    req_slot_id_param = 0;
+    status = libspdm_send_receive_finish(spdm_context, session_id, req_slot_id_param);
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(
+        libspdm_secured_message_get_session_state(
+            spdm_context->session_info[0].secured_message_context),
+        LIBSPDM_SESSION_STATE_ESTABLISHED);
+
+    secured_message_context = session_info->secured_message_context;
+
+    assert_memory_equal((const void *)secured_message_context->master_secret.master_secret,
+                        (const void *)m_libspdm_zero_buffer, sizeof(m_libspdm_zero_buffer));
+    free(data);
+}
+
+/**
+ * Test 25: Set HANDSHAKE_IN_THE_CLEAR_CAP to 0 , The ResponderVerifyData field is present.
+ * Expected behavior: client returns a Status of RETURN_DEVICE_ERROR.
+ **/
```

Comment:
Here `RETURN_DEVICE_ERROR` is not aligned with `LIBSPDM_STATUS_INVALID_MSG_SIZE` at line 4014

Reply:
Thanks , I have modified this place